src/ignore.rs

```diff
@@ -48,12 +54,23 @@ impl GitIgnore {
         .iter()
         .collect();
 
+        let detectors = vec![Detector {
+            matches: Box::new(|files| {
+                files
+                    .into_iter()
+                    .find(|f| f.file_name() == "Cargo.toml")
```

Comment:
I think this would benefit from being a `Vec<String>` instead of a single file, some things can have multiple names for the same thing. For example Gradle: you can use either `build.gradle` and `build.gradle.kts`, so something like `find(|f| self.aliases.contains(f.file_name())` (ignore the name for `self.aliases`, can't come up with anything quickly).